### PR TITLE
Add TypeScript Linter

### PR DESCRIPTION
**Pull request by mattruggio**
This PR will add TSLint and also address all issues.  I used the default settings with only one single-file override:

- util/logger.ts references console.log, which the linter does not like.  In the future I will look to replace this with an actual first-class logger.  For now, I marked this as safe.

I also bumped the version to 2.0.0 due to a breaking change:

- logic.ts exports a top level function (evaluate) instead of a top-level module with the function inside.  The linter did not like this so I took a second look at options.  The take-away here is that removing the module removes un-necessary indirection (there really is no use of the namespace in this context.)

A couple things I added:

1. updated the dependencies, they contained some development-only ones.  
2. Added HipChat notifications for BMP internal.  The token is encrypted via[ travis-ci recommendation](https://docs.travis-ci.com/user/notifications/#configuring-hipchat-notifications)

### CODE REVIEW

File: lib/util/logger.ts [hunk 1]
```diff
@@ -5,20 +5,20 @@
...
+export function off() {
+  turnedOn = false;
+}
 
-  export function log(msg) {
-    if (turnedOn && console && console.log) {
-      console.log(`[LOG] ${msg}`);
-    }
+export function log(msg) {
+  // tslint:disable-next-line:no-console
```
Comment (gerryster) [inline lib/util/logger.ts]: Probably worth adding a TODO to extract a logger component.

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
+    ],
+    "jsRules": {},
+    "rules": {},
+    "rulesDirectory": []
+}
```
Comment (gerryster) [inline tslint.json]: Missing and end of file newline here.

Comment (mattruggio) [reply tslint.json]: Added.

File: lib/util/logger.ts (hunk 1 above)
Comment (mattruggio) [reply lib/util/logger.ts]: Added.

**Pull request merged**